modules/aws/master/master.tf

```diff
@@ -18,69 +15,85 @@ resource "aws_instance" "master" {
 
   associate_public_ip_address = false
   source_dest_check           = false
-  subnet_id                   = "${var.master_subnet_ids[count.index]}"
-  vpc_security_group_ids      = ["${aws_security_group.master.id}"]
+
+  subnet_id              = "${var.master_subnet_ids[count.index]}"
+  vpc_security_group_ids = ["${aws_security_group.master.id}"]
 
   root_block_device = {
     volume_type = "${var.volume_type}"
     volume_size = "${var.volume_size_root}"
   }
 
-  user_data = "${data.ignition_config.s3.rendered}"
+  user_data = "${element(data.ignition_config.s3.*.rendered, count.index)}"
 
   tags = "${merge(
     local.common_tags,
     map(
       "Name", "${var.cluster_name}-master${count.index}"
     )
   )}"
+
+  # we ignore changes, to avoid rolling all masters at once
+  # update is done via tainting masters
+  lifecycle {
+    ignore_changes = ["*"]
```

Comment:
If we are planning to taint and then apply to protect the rollout maybe we can include here  create_before_destroy = true, to avoid the lose of one master if the creation fails.

Extra point would be create a local-exec script in this resource to check the availability of the resource before destroying the old one. If the resource is created but the health script fail won't be destroyed either.

Reply:
👍 Good level of automation. I like that.

Reply:
That's a good idea, But I don't want to add more complexity to this story, this would be a nice fit for a separate issue we can address later, can you create an issue to sig-infra @ferrandinand

Reply:
ok, let´s do that.

Reply:
Agree to address this in a different PR.